src/v/rpc/connection_cache.cc

```diff
@@ -100,20 +100,53 @@ ss::future<> connection_cache::stop() {
 }
 
 ss::shard_id connection_cache::shard_for(
-  model::node_id self,
+  model::node_id src_node,
   ss::shard_id src_shard,
-  model::node_id n,
+  model::node_id dest_node,
   ss::shard_id total_shards) const {
     if (total_shards <= _max_connections) {
         return src_shard;
     }
 
     // NOLINTNEXTLINE
-    size_t h = 805306457;
-    boost::hash_combine(h, jump_consistent_hash(src_shard, _max_connections));
-    boost::hash_combine(h, std::hash<model::node_id>{}(n));
-    boost::hash_combine(h, std::hash<model::node_id>{}(self));
-    // use self node id to shift jump_consistent_hash_assignment
-    return jump_consistent_hash(h, total_shards);
+    uint64_t connection_hash = 201326611; // magic prime
+    boost::hash_combine(connection_hash, std::hash<model::node_id>{}(src_node));
+    boost::hash_combine(
+      connection_hash, std::hash<model::node_id>{}(dest_node));
+
+    // Find the largest ring_size s.t. ring_size <= total_shards
+    // and ring_size % connections_per_node == 0
+    uint64_t remainder = total_shards % _max_connections;
+    uint64_t ring_size = total_shards - remainder;
+
+    // Vary the starting shard of the ring depending on the connection hash.
```

Comment:
So this is the actual "consistent hash" right? In that case I don't understand why we use remainder below for the bucket size.

Reply:
Can you please add comments/links to where/how those magic primes (there are more than one!) came about to be? Is there a paper or is it on OSS implementation from elsewhere?

Reply:
I guess boost::hash_combine does not do very good mixing? I wonder if `absl::HashOf` is better. It seems a bit weird to mix in a "magic prime" which seem to be optimized to be used as the second argument to `%` in a hash table, and not to be used as the seed for a mixing function like this.

Reply:
It's worth noting that these primes aren't new: the original `shard_for` used a list of them from the same source too: it's just not that clear from the diff because it moved files.

Reply:
`absl::HashOf` will work much better. The magic hash is mainly there since `std::hash` is just the identity function for integers which doesn't result in a very good distribution of connections.